mesonbuild/interpreterbase.py

```diff
@@ -369,14 +369,19 @@ def evaluate_plusassign(self, node):
     def evaluate_indexing(self, node):
         assert(isinstance(node, mparser.IndexNode))
         iobject = self.evaluate_statement(node.iobject)
-        if not isinstance(iobject, list):
-            raise InterpreterException('Tried to index a non-array object.')
+        if not (isinstance(iobject, list) or hasattr(iobject, 'index_protocol')):
+            raise InterpreterException(
+                'Tried to index an object that doesn\'t support indexing.')
```

Comment:
Then this can just be `hasattr(iobject, '__getitem__')`, which will work for all indexable objects.